Packs/VMwareWorkspaceONEUEM/Integrations/VMwareWorkspaceONEUEM/VMwareWorkspaceONEUEM.py

```diff
@@ -0,0 +1,644 @@
+    500: "The server encountered an internal error for VMWare Workspace ONE UEM "
+         "and was unable to complete your request."
+}
+ARG_TO_PARAM_OWNERSHIP = {
+    "corporate owned": "C",
+    "employee owned": "E",
+    "shared": "S",
+    "undefined": "undefined"
+}
+REVERSED_ARG_TO_PARAM_OWNERSHIP = {
+    'C': 'Corporate owned',
+    'E': 'Employee owned',
+    'S': 'Shared',
+    'Undefined': 'Undefined'
+}
+MESSAGES = {
+    "NO_RECORDS_FOUND": "No {} record(s) found for the given argument(s).",
+    "INVALID_PAGE_SIZE": "Argument page_size should be greater than 1.",
+    "INVALID_PAGE": "Argument page should be greater than 0.",
+    "INVALID_OWNERSHIP": "Argument ownership should be one of the following: "
+                         "Corporate owned, Employee owned, Shared, or Undefined.",
+    "INVALID_SORT_ORDER": "Argument sort_order should be one of the following: ASC, or DESC.",
+    "REQUIRED_ARGUMENT": "{} is a required argument.",
+    "INVALID_COMPLIANCE_STATUS": "Argument compliance_status should be one of the following: true, or false."
+}
+CONSTANT_STRING = {
+    "DEVICE_FRIENDLY": "Device Friendly Name",
+    "SERIAL_NUM": "Serial Number",
+    "MAC_ADDR": "MAC Address",
+    "COMPLIANCE_STATUS": "Compliance Status",
+    "USER_EMAIL": "User Email Address",
+    "LAST_SEEN": "Last Seen (In UTC)"
+}
+OUTPUT_PREFIX = {
+    "SEARCH_DEVICES": "VMwareWorkspaceONEUEM.Device",
+    "LIST_OS_UPDATES": "VMwareWorkspaceONEUEM.OSUpdate",
+    "GET_DEVICE": "VMwareWorkspaceONEUEM.Device",
+}
+''' CLIENT CLASS '''
+
+
+class Client(BaseClient):
+    """Client class to interact with the service API
+
+    This Client implements API calls, and does not contain any XSOAR logic.
+    Should only do requests and return data.
+    It inherits from BaseClient defined in CommonServer Python.
+    Most calls use _http_request() that handles proxy, SSL verification, etc.
+    For this  implementation, no special attributes defined
+    """
+
+    def __init__(self, username, password, base_url, headers, verify=True, proxy=False):
+        """
+        Store username and password for authentication.
+
+        :type username: ``string``
+        :param username: username of salesforce account
+
+        :type password: ``string``
+        :param password: password of salesforce account
+
+        :type base_url: ``string``
+        :param base_url: service API url.
+
+        :type headers: ``dict``
+        :param headers: The request headers, for example: {'Accept`: `application/json`}.
+
+        :type verify: ``bool``
+        :param verify: SSL verification is handled out of the box.
+
+        :type proxy: ``bool``
+        :param proxy: system proxy is handled out of the box.
+        """
+
+        super().__init__(base_url=base_url, auth=(username, password), headers=headers, verify=verify, proxy=proxy)
+
+    def http_request(self, method: str, url_suffix: str = '', full_url: str = None, params: Dict = None,
```

Comment:
@Shellyber 
To avoid passing the same parameters such as `resp_type`, `ok_codes`, and `exception_hadlers` during each _http_request, we decided to keep it abstract by overriding the base client function.

Reply:
You can wrap the http request we are using. 
for example: 
```
def https_request(*args, **kwargs):
    kwargs['ok_codes'] = (200, 204)
    super().http_request(*args, **kwargs)
```
No need to write it again.